### PR TITLE
[Windows]wsl for windows server

### PR DESCRIPTION
**Pull request by 123lzxm**
Signed-off-by: Yanan Shen <yanans@vmware.com>

### CODE REVIEW

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 1]
```diff
@@ -41,6 +62,7 @@
   when:
     - win_powershell_cmd_output.rc is defined
     - win_powershell_cmd_output.rc != 0
+    - (guest_os_product_type | lower == 'client')
```
Comment (Tomorrow9) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:65]: Only check return code for client?

Comment (123lzxm) [reply windows/wsl_distro_install_uninstall/install_wsl.yml]: Sorry for late response. 
Change the condition to only skip result check for lower version of windows server, whose RC=1 for the WSL installation task even the WSL and distribution are installed successfully.

**Comment by 123lzxm** [comment]
```
+----------------------------------------------------------------------+
| Name                      | wsl_win_serve_2022_ga                    |
+----------------------------------------------------------------------+
| IP                        |                       |
+----------------------------------------------------------------------+
| Guest OS Distribution     | Microsoft Windows Server 2022 Datacenter |
|                           | 10.0.20348.0 64-bit                      |
+----------------------------------------------------------------------+
| Hardware Version          | vmx-20                                   |
+----------------------------------------------------------------------+
| VMTools Version           |                                          |
+----------------------------------------------------------------------+
| Config Guest Id           | windows2019srvNext_64Guest               |
+----------------------------------------------------------------------+
| GuestInfo Guest Id        |                                          |
+----------------------------------------------------------------------+
| GuestInfo Guest Full Name |                                          |
+----------------------------------------------------------------------+
| GuestInfo Guest Family    |                                          |
+----------------------------------------------------------------------+
| GuestInfo Detailed Data   |                                          |
+----------------------------------------------------------------------+
+------------------------------------------------------+
| Name                         |   Status  | Exec Time |
+------------------------------------------------------+
| deploy_vm                    | * Skipped | 00:00:00  |
| wsl_distro_install_uninstall |   Passed  | 00:09:35  |
+------------------------------------------------------+
```

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 2]
```diff
@@ -9,25 +9,54 @@
...
+
+    - name: "Install WSL and distribution"
+      ansible.windows.win_shell: "wsl --install -d {{ wsl_distribution_name }} | Out-File -FilePath {{ wsl_file_path_win }}\\wsl_install.txt"
+      register: win_powershell_cmd_output
+      ignore_errors: true
+      delegate_to: "{{ vm_guest_ip }}"
+      ignore_unreachable: true
+      become: true
+      become_method: runas
+      become_user: Administrator
```
Comment (keirazhang) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:50]: The username required for Windows Server is `Administrator`, why here needs to become `Administrator`?

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 3]
```diff
@@ -41,6 +70,7 @@
   when:
     - win_powershell_cmd_output.rc is defined
     - win_powershell_cmd_output.rc != 0
+    - not skip_set_wsl_ver
```
Comment (keirazhang) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:73]: Where is the default value of skip_set_wsl_ver set? I only saw it's set as true when `guest_os_build_num | int <= 23048` and `guest_os_product_type | lower == 'server'`. If it is not initialized, you'd better to add checking about skip_set_wsl_ver is defined or not.

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 4]
```diff
@@ -56,16 +86,40 @@
...
       - wsl_default_version | int == 2
     fail_msg: "WSL default version is: {{ wsl_default_version }}, not the expected version: 2"
+  when: not skip_set_wsl_ver
+
+- name: "Validate the WSL version"
+  ansible.builtin.assert:
+    that:
+      - wsl_default_version
+      - wsl_default_version | int == 1
+    fail_msg: "WSL default version is: {{ wsl_default_version }}"
```
Comment (keirazhang) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:124]: It's better to add "not the expected version: 1" in the fail message as well.

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 5]
```diff
@@ -9,25 +9,58 @@
 # and the support of inbox version is removed from Windows build 25267, the test case will install inbox version for
 # build lower than 25267 until the known issue is resolved.
 #
-- name: "Set installation command for inbox version of WSL"
+- name: "Initialize some facts"
```
Comment (Tomorrow9) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:12]: This name part can be more specific.

File: windows/wsl_distro_install_uninstall/install_wsl.yml [hunk 6]
```diff
@@ -9,25 +9,58 @@
...
-    (guest_os_build_num | int >= 25267) or
-    (guest_os_build_num | int < 22000)
+- name: "Install WSL for Windows Client"
+  block:
+    - name: "Set installation command for inbox version of WSL"
+      ansible.builtin.set_fact:
+        wsl_install_cmd: "wsl --install --inbox -d {{ wsl_distribution_name }} | Out-File -FilePath {{ wsl_file_path_win }}\\wsl_install.txt"
+      when:
+        - guest_os_build_num | int < 25267
+        - guest_os_build_num | int > 22000
```
Comment (Tomorrow9) [inline windows/wsl_distro_install_uninstall/install_wsl.yml:23]: How about guest_os_build_num | int == 22000 ?

File: windows/wsl_distro_install_uninstall/install_wsl.yml (hunk 5 above)
Comment (123lzxm) [reply windows/wsl_distro_install_uninstall/install_wsl.yml:12]: Updated.

File: windows/wsl_distro_install_uninstall/install_wsl.yml (hunk 6 above)
Comment (123lzxm) [reply windows/wsl_distro_install_uninstall/install_wsl.yml:23]: Yes, missing the process for 22000. Fixed. Thanks.

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**